src/Chapter3.hs

```diff
@@ -480,6 +528,89 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = Castle
+  {
+    name :: String
+  }
+
+data IsCastle = 
+  Yes Castle |
+  No
+
+data WhichBuilding = 
+  Chruch |
+  Library
+
+data People = 
+  One Int |
+  Two Int |
+  Three Int |
+  Four Int
+
+data House = House
+  {
+    people :: People
+  }
+
+-- How to represent the Wall object?
+data City = City
+  {
+    castle :: IsCastle,
+    wall :: Bool,
+    building :: WhichBuilding,
+    houses :: [House]
+  }
+
+buildCastle :: String -> City -> City
+buildCastle newName city =
+  newCity
+  where
+    newCity :: City
+    newIsCastle :: IsCastle
+    newCastle :: Castle
+    newCastle = Castle 
+      {
+        name = newName
+      }
+    newIsCastle = Yes newCastle
+    newCity = city {castle = newIsCastle}
```

Comment:
You usually don't separate function type signature and the function body. That would make code more readable 🙂